packages/react/src/components/navigation/progress/Stepper/index.tsx

```diff
@@ -1,17 +1,28 @@
-import React, { memo, ReactNode } from "react";
+import React, { memo } from "react";
 import styled from "styled-components";
 import { border, BorderProps, color, ColorProps, space, SpaceProps } from "styled-system";
 import CheckAlone from "@ledgerhq/icons-ui/react/CheckAloneMedium";
 import CloseMedium from "@ledgerhq/icons-ui/react/CloseMedium";
 import Text from "../../../asorted/Text";
 import Flex from "../../../layout/Flex";
 
-type Label = string | ReactNode;
+/**
+ * The state of a progress bar step.
+ */
+export type StepState = "pending" | "current" | "completed" | "errored";
+
+type LabelType =
+  | string
+  | React.ComponentType<{ state: StepState }>
+  | ((props: { state: StepState }) => JSX.Element);
```

Comment:
I think that the `React.ComponentType<{ state: StepState }>` type encompass `((props: { state: StepState }) => JSX.Element)`.